sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -107,3 +107,21 @@ def test_hasher_zeros():
     # Assert that no zeros are materialized in the output.
     X = FeatureHasher().transform([{'foo': 0}])
     assert_equal(X.data.shape, (0,))
+
+
+def test_hasher_non_negative():
+    raw_X = [["foo", "bar", "baz"]]
+
+    def it():  # iterable
```

Comment:
why not use `raw_X` directly?

And convention would use `X` for input, and `Xt` for output.